packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -194,4 +195,111 @@ describe("R11s Socket Tests", () => {
 			"Error internal code should be R11sServiceClusterDrainingErrorCode",
 		);
 	});
+
+	it("Socket error with Data Corruption error", async () => {
+		const errorToThrow = {
+			code: 500,
+			message: "Data corruption detected",
+			retryAfterMs: 0,
+			internalErrorCode: "DataCorruption",
+			errorType: FluidErrorTypes.dataCorruptionError,
+			canRetry: false,
+		};
+		const errorEventName = "connect_document_success";
+		socket = new ClientSocketMock({
+			connect_document: { eventToEmit: errorEventName },
+		});
+
+		const connection = await mockSocket(socket as unknown as Socket, async () =>
+			documentService.connectToDeltaStream(client),
+		);
+		let error: IAnyDriverError | undefined;
+		connection.on("disconnect", (reason: IAnyDriverError) => {
+			error = reason;
+		});
+
+		// Send Data Corruption error
+		socket.sendErrorEvent(errorToThrow);
```

Comment:
I don't think this is doing what we want.

My understanding is that the `socketMock.sendErrorEvent()` is mocking an error _from the server_. The change in this PR is supposed to be testing that the server receives an error from the client before it disconnects.